packages/core/src/components/fields/ArrayField.tsx

```diff
@@ -0,0 +1,859 @@
+    if (isFixedItems(schema) && allowAdditionalItems(schema)) {
+      itemSchema = schema.additionalItems as RJSFSchema;
+    }
+    // Cast this as a T to work around schema utils being for T[] caused by the FieldProps<T[], F> call on the class
+    return schemaUtils.getDefaultFormState(itemSchema) as unknown as T;
+  };
+
+  /** Callback handler for when the user clicks on the add button. Creates a new row of keyed form data at the end of
+   * the list, adding it into the state, and then returning `onChange()` with the plain form data converted from the
+   * keyed data
+   *
+   * @param event - The event for the click
+   */
+  onAddClick = (event: MouseEvent) => {
+    if (event) {
+      event.preventDefault();
+    }
+
+    const { onChange } = this.props;
+    const { keyedFormData } = this.state;
+    const newKeyedFormDataRow: KeyedFormDataType<T> = {
+      key: generateRowId(),
+      item: this._getNewFormDataRow(),
+    };
+    const newKeyedFormData = [...keyedFormData, newKeyedFormDataRow];
+    this.setState(
+      {
+        keyedFormData: newKeyedFormData,
+        updatedKeyedFormData: true,
+      },
+      () => onChange(keyedToPlainFormData(newKeyedFormData))
+    );
+  };
+
+  /** Callback handler for when the user clicks on the add button on an existing array element. Creates a new row of
+   * keyed form data inserted at the `index`, adding it into the state, and then returning `onChange()` with the plain
+   * form data converted from the keyed data
+   *
+   * @param index - The index at which the add button is clicked
+   */
+  onAddIndexClick = (index: number) => {
+    return (event: MouseEvent) => {
+      if (event) {
+        event.preventDefault();
+      }
+      const { onChange } = this.props;
+      const { keyedFormData } = this.state;
+      const newKeyedFormDataRow: KeyedFormDataType<T> = {
+        key: generateRowId(),
+        item: this._getNewFormDataRow(),
+      };
+      const newKeyedFormData = [...keyedFormData];
+      newKeyedFormData.splice(index, 0, newKeyedFormDataRow);
+
+      this.setState(
+        {
+          keyedFormData: newKeyedFormData,
+          updatedKeyedFormData: true,
+        },
+        () => onChange(keyedToPlainFormData(newKeyedFormData))
+      );
+    };
+  };
+
+  /** Callback handler for when the user clicks on the remove button on an existing array element. Removes the row of
+   * keyed form data at the `index` in the state, and then returning `onChange()` with the plain form data converted
+   * from the keyed data
+   *
+   * @param index - The index at which the remove button is clicked
+   */
+  onDropIndexClick = (index: number) => {
+    return (event: MouseEvent) => {
+      if (event) {
+        event.preventDefault();
+      }
+      const { onChange, errorSchema } = this.props;
+      const { keyedFormData } = this.state;
+      // refs #195: revalidate to ensure properly reindexing errors
+      let newErrorSchema: ErrorSchema<T>;
```

Comment:
Can this be a <T[]> now?

Reply:
Nope:
<img width="824" alt="Screen Shot 2022-08-12 at 9 19 01 AM" src="https://user-images.githubusercontent.com/51679588/184399966-5eb86674-a417-4106-8770-89a927044446.png">